sharing_portal/views.py

```diff
@@ -949,3 +966,109 @@ def create_supplemental_project_if_needed(request, artifact, project):
             artifact_uuid=artifact["uuid"], project=created_project
         )
         daypass_project.save()
+
+
+@check_edit_permission
+@with_trovi_token
+@login_required
+def create_git_version(request, artifact):
+    errors = False
+    if request.method == "POST":
+        remote_url = request.POST.get("gitRemote")
+        git_ref = request.POST.get("gitRef")
+        # Validate git ref
+        for item in ls_remote(remote_url):
+            if item[0] == git_ref:
+                break
+        else:
+            messages.add_message(
+                request,
+                messages.ERROR,
+                "Either invalid git reference specified, or invalid remote URL",
+            )
+            errors = True
+        if not errors:
+            trovi.create_version(
+                request.session.get("trovi_token"),
+                artifact["uuid"],
+                f"urn:trovi:contents:git:{remote_url}@{git_ref}",
+            )
+            messages.add_message(
+                request,
+                messages.SUCCESS,
+                "Successfully created artifact version",
+            )
+            return HttpResponseRedirect(
+                reverse("sharing_portal:detail", args=[artifact["uuid"]])
+            )
+    template = loader.get_template("sharing_portal/create_git_version.html")
+    return HttpResponse(template.render({}, request))
+
+
+# Login required to prevent someone potentially abusing this
+@login_required
+def get_remote_data(request):
+    remote_url = request.GET.get("remote_url")
+    return JsonResponse({"result": ls_remote(remote_url)})
+
+
+def ls_remote(remote_url):
+    remote_url = remote_url.strip()
+    res = subprocess.run(["git", "ls-remote", remote_url], capture_output=True)
```

Comment:
Is there any way in which this can fail? I see `check` is False, which means it won't throw. It may be worthwhile to log a warning if this fails and fall back to returning an empty list rather than silently failing.

Reply:
There are a lot of issues that may arise. I'm unsure of exactly how dangerous the output of `ls-remote` is and if it is safe to display it to the user.